x-pack/platform/plugins/shared/inference/server/chat_complete/anonymization/execute_regex_rule_task.ts

```diff
@@ -6,35 +6,48 @@
  */
 
 import type { RegexAnonymizationRule } from '@kbn/inference-common';
-import type { Anonymization } from '@kbn/inference-common';
-import { getEntityMask } from './get_entity_mask';
-import type { AnonymizationState } from './types';
+import type { DetectedMatch } from './types';
 
```

Comment:
Suggestion to add some documentation/comments here for the function. Probably good to note the order of the rule being passed to the `rules` array is important (earlier rule takes precedent over later rule).